File: product-front/src/services/ProductService.js
import api from './api';

const getAllProducts = () => {
    return api.get('/products');
};

const getProductById = (id) => {
    return api.get(`/products/${id}`);
};

const createProduct = (product) => {
    return api.post('/products', product);
};

const updateProduct = (id, product) => {
    return api.put(`/products/${id}`, product);
};

const deleteProduct = (id) => {
    return api.delete(`/products/${id}`);
};

const ProductService = {
    getAllProducts,
    getProductById,
    createProduct,
    updateProduct,
    deleteProduct
};

export default ProductService; 

File: product-back/src/main/java/sotware/hahn/product/service/ProductService.java
package sotware.hahn.product.service;

import sotware.hahn.product.dto.ProductRequestDto;
import sotware.hahn.product.dto.ProductResponseDto;

import java.util.List;

public interface ProductService {
    List<ProductResponseDto> getAllProducts();
    ProductResponseDto getProductById(Long id);
    ProductResponseDto createProduct(ProductRequestDto productRequestDto);
    ProductResponseDto updateProduct(Long id, ProductRequestDto productRequestDto);
    void deleteProduct(Long id);
} 